va/validation-authority.go

```diff
@@ -54,6 +54,26 @@ type verificationRequestEvent struct {
 
 // Validation methods
 
+// parseDNSError checks the error returned from Lookup... methods and tests if the error
+// was an underlying net.OpError or an error caused by resolver returning SERVFAIL or other
+// invalid Rcodes.
+func (va ValidationAuthorityImpl) parseDNSError(err error, challenge core.Challenge) core.Challenge {
```

Comment:
Also, there's no reason for this to be a method on the `ValidationAuthorityImpl` object.  Just make it a function.